src/coreclr/src/jit/morph.cpp

```diff
@@ -2837,10 +2846,11 @@ void Compiler::fgInitArgInfo(GenTreeCall* call)
         {
             maxRegArgs = 0;
         }
-
+#ifdef UNIX_X86_ABI
```

Comment:
why do we need this change?

Reply:
This change was to preserve the previous behavior. We used to do this on both x86 platforms. For consistency with the previous behavior (since I don't have a unix x86 test setup and we don't run it in CI), I wanted to keep this block active for Unix x86.